packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,461 @@
+   * HTML Attribute: modalType
+   */
+  @attr({ attribute: 'modal-type' })
+  public modalType: DrawerModalType = DrawerModalType.modal;
+
+  /**
+   * The ID of the element that labels the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-labelledby
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * The ID of the element that describes the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-describedby
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * Sets the type of the drawer (overlay/inline).
+   * @public
+   * @remarks
+   * HTML Attribute: type
+   * @defaultValue overlay
+   */
+  @attr
+  public type?: DrawerType = DrawerType.overlay;
+
+  /**
+   * Sets the position of the drawer (start/end).
+   * @public
+   * @remarks
+   * HTML Attribute: position
+   * @defaultValue end
+   */
+  @attr
+  public position?: DrawerPosition;
+
+  /**
+   * Sets the size of the drawer (small/medium/large).
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   * @defaultValue medium
+   */
+  @attr({ attribute: 'size' })
+  public size?: DrawerSize;
+
+  /**
+   * Sets the open state of the drawer
+   * @public
+   * @remarks
+   * HTML Attribute: open
+   * @defaultValue false
+   */
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
+
+  /**
+   * Emits a 'cancel' event.
+   * @internal
+   */
+  public cancel(): void {
+    this.$emit('cancel');
+  }
+
+  /**
+   * Handles changes to the `open` property.
+   * @public
+   */
+  public openChanged(oldValue: boolean, newValue: boolean): void {
+    if (newValue !== oldValue) {
+      if (newValue && !oldValue) {
+        this.show();
+      } else if (!newValue && oldValue) {
+        this.close();
+      }
+    }
+    this.$emit('onOpenChange', { open: this.open });
+  }
+
+  /**
+   * @public
+   * Method called when the 'modalType' attribute changes
+   */
+  public typeChanged(oldValue: DrawerType, newValue: DrawerType): void {
+    if (newValue !== oldValue) {
+      if (newValue == DrawerType.inline) {
+        this.modalType = DrawerModalType.nonModal;
+      }
+    }
+  }
+
+  /**
+   * @public
+   * Method called when the 'modalType' attribute changes
+   */
+  public modalTypeChanged(oldValue: DrawerModalType, newValue: DrawerModalType): void {
+    if (newValue !== oldValue) {
+      if (newValue == DrawerModalType.nonModal) {
+        this.trapFocus = false;
+      } else {
+        this.trapFocus = true;
+      }
+    }
+  }
+
+  /**
+   * Sets the component state based on the `open` property.
+   * @public
+   */
+  public setComponent(): void {
+    if (this.modalType == DrawerModalType.nonModal) {
+      this.trapFocus = false;
+    } else {
+      this.trapFocus = true;
+    }
+    if (this.open) {
+      this.show();
+    }
+  }
+
+  /**
+   * Opens the drawer if it is currently closed.
+   * @public
+   */
+  public show(): void {
```

Comment:
Any reason why you selected: show() and close() instead of open/close show/hide?
Show may imply that it's on the DOM, only hidden. Is this a mix of conventions - one for Drawer, the other for Dialog?

Reply:
Dialog is using `hide()` and `show()` as well for the method names. It might not have been that way when you left the comment. I'm using the native dialog component for both the `drawer` and `dialog` components. The idea was to match the semantics use in the native dialog which has three methods `hide`, `show`, and `showModal`.